ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{deneb::blob_sidecar::Blob, primitives::Bytes32, ssz::prelude::*};
+pub use c_kzg::KzgSettings;
+use std::fmt;
+use thiserror::Error;
+
+pub const BYTES_PER_FIELD_ELEMENT: usize = 32;
+pub const BYTES_PER_COMMITMENT: usize = 48;
+pub const BYTES_PER_PROOF: usize = 48;
+
+pub type VersionedHash = Bytes32;
+pub type FieldElement = Bytes32;
+pub type KzgCommitment = ByteVector<BYTES_PER_COMMITMENT>;
+pub type KzgProof = ByteVector<BYTES_PER_PROOF>;
+
+#[derive(Debug, Error)]
+pub struct Error(c_kzg::Error);
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // Implement custom formatting logic here
+        write!(f, "Custom error: {:?}", self.0)
+    }
+}
+
+impl From<c_kzg::Error> for Error {
+    fn from(inner_error: c_kzg::Error) -> Self {
+        Error(inner_error)
+    }
+}
+
+pub struct ProofAndEvaluation {
+    pub proof: KzgProof,
+    pub evaluation: Bytes32,
+}
+
+pub fn blob_to_kzg_commitment<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgCommitment, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref())?;
+
+    let commitment = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).expect("Correct size");
```

Comment:
lowercase better here
```suggestion
    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).expect("correct size");
```